os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c

```diff
@@ -1266,6 +1282,7 @@ void app_start(void)
 	VectorTableOverride();
 
 	app_section_init();
+	app_RAM_data_copy();
```

Comment:
I have verified on latest code base, the global data from flash (example : kregionx_start , kregionx_size) is copied to RAM properly. So the current change of app_RAM_data_copy is not required as you mentioned its done in BL.

Can you remove os_heap_init() function call below and its definition, as kregionx_start is set at up_allocate_kheap() in os_start common codes. It is duplicate otherwise to call below.

Reply:
Just to confirm, Realtek os_heap_init() used the free stack info on .ld. So it will allocate all used stack as free heap.
Are up_allocate_kheap() does the same, or it is allocated base on user config file?